Add caption and name options to output blocks

Madricas opened the pull request:
I would like to be able to caption and reference `program-output` and `command-output` block like `literalinclude` directives.

As such, this PR is heavily inspired from what is already done for `literalinclude`.
It adds the `:name:` option to be able to reference the output block from elsewhere, and the `:caption:` option to set a caption to the output block.

Thanks in advance for looking at it.

jamadden commented:
Thanks for this PR! I don’t see any reason not to add this functionality. 

This is a great start. The PR will need a few things before it can be merged:

- [ ] Thorough tests
- [ ] compatible syntax
- [ ] No new linter errors
- [ ] documentation, plus a note in CHANGES.rst

Let me know if I can assist with any of those things.

Madricas commented:
Hi!

I have documented the options in the manual and added a note in the `CHANGES.rst`.
I have also added some tests which required to modify a bit the `TestDirective`'s method `assert_output`.

The linter detects an error in the `src/sphinxcontrib/programoutput/tests/__init__.py` for Python3. I did not modify this file, so was this an existing issue ?

I let you review the code and make any modification.
Thanks.

jamadden commented:
> so was this an existing issue ?

Yes, an update to pylint added a new warning. I've taken care of it.

Thanks!
